ext/symengine/symengine_utils.c

```diff
@@ -28,6 +30,22 @@ void sympify(VALUE operand2, basic_struct *cbasic_operand2) {
             basic_free_stack(den_basic);
             break;
 
+        case T_COMPLEX:
+            real = rb_funcall(operand2, rb_intern("real"), 0, NULL);
+            imag = rb_funcall(operand2, rb_intern("imaginary"), 0, NULL);
+
+            basic_struct *real_basic = basic_new_heap();
+            basic_struct *imag_basic = basic_new_heap();
+
+            sympify(real, real_basic);
+            sympify(imag, imag_basic);
+
+            complex_set(cbasic_operand2, real_basic, imag_basic);
```

Comment:
`complex_set` requires integers or rationals, but Ruby complex can have any real number as real and imaginary parts.
Best way is to do `cbasic_operand2 = real_basic + basic_const_I * imag_basic `